mobilecoind/src/conversions.rs

```diff
@@ -315,6 +365,84 @@ mod test {
         assert_eq!(rust, UnspentTxOut::try_from(&proto).unwrap());
     }
 
+    // Test the From<&MemoPayload> for api::DecodedMemo implementation
+    #[test]
+    fn test_memo_conversion() {
+        let mut rng: StdRng = SeedableRng::from_seed([1u8; 32]);
+
+        let alice = AccountKey::new(
+            &RistrettoPrivate::from_random(&mut rng),
+            &RistrettoPrivate::from_random(&mut rng),
+        );
+        let alice_cred = SenderMemoCredential::from(&alice);
+        let alice_hash = alice_cred.address_hash.clone();
+
+        let bob = AccountKey::new(
+            &RistrettoPrivate::from_random(&mut rng),
+            &RistrettoPrivate::from_random(&mut rng),
+        );
+        let bob_addr = bob.default_subaddress();
+
+        let tx_public_key = CompressedRistrettoPublic::from_random(&mut rng);
+
+        let memo1 = UnusedMemo {};
+        let decoded = decode_memo(&MemoPayload::from(memo1));
+        assert!(!decoded.has_authenticated_sender_memo());
+        assert!(!decoded.has_unknown_memo());
+
+        let memo2 =
+            AuthenticatedSenderMemo::new(&alice_cred, bob_addr.view_public_key(), &tx_public_key);
+        let decoded = decode_memo(&MemoPayload::from(memo2));
+        assert!(decoded.has_authenticated_sender_memo());
+        assert!(!decoded.has_unknown_memo());
```

Comment:
nit: this is a one of, if we had 10 variants would we want be checking the negative cases of all 9 alternatives?  Suggest fixing other places as well if this is changed.
```suggestion
        assert!(decoded.has_authenticated_sender_memo());
```

Reply:
yeah agree. if `.has_X` returns true, the API contract for protobuf lib says that `.has_Y` must be false, so these negative assertions are not testing our code, they are testing the protobuf lib, which probably isn't very useful to us.